geode-lucene/src/main/java/org/apache/geode/cache/lucene/internal/LuceneIndexForPartitionedRegion.java

```diff
@@ -78,17 +87,9 @@ protected RepositoryManager createRepositoryManager() {
     // create PR fileAndChunkRegion, but not to create its buckets for now
     final String fileRegionName = createFileRegionName();
     PartitionAttributes partitionAttributes = dataRegion.getPartitionAttributes();
-
-
-    // create PR chunkRegion, but not to create its buckets for now
-
-    // we will create RegionDirectories on the fly when data comes in
-    HeterogeneousLuceneSerializer mapper = new HeterogeneousLuceneSerializer(getFieldNames());
-    PartitionedRepositoryManager partitionedRepositoryManager =
-        new PartitionedRepositoryManager(this, mapper);
     DM dm = this.cache.getInternalDistributedSystem().getDistributionManager();
     LuceneBucketListener lucenePrimaryBucketListener =
-        new LuceneBucketListener(partitionedRepositoryManager, dm);
+        new LuceneBucketListener((PartitionedRepositoryManager) partitionedRepositoryManager, dm);
```

Comment:
I made createLuceneListenerAndFileChunkRegion method take in PartitionedRepositoryManager
which removed all the casting in the LuceneIndexForPartitionRegion.java but I would like a suggestion on how to remove the casting in createLuceneListenersAndFileChunkRegions(  (PartitionedRepositoryManager) repositoryManager); in the LuceneIndexImpl.java file. As the repositoryManager is being created by createRepositoryManager() which creates  repository manager for both RawLuceneIndex as well as LuceneIndexForPartitionRegion